packages/squiggle-lang/src/rescript/Distributions/PointSetDist/PointSetDist_Scoring.res

```diff
@@ -1,46 +1,185 @@
-    ~predictionPdf: float => float,
-    ~answer: float,
+module WithScalarAnswer = {
+  let score' = (~estimatePdf: float => float, ~answer: float): result<float, Operation.Error.t> => {
+    let density = answer->estimatePdf
+    if density < 0.0 {
+      Operation.PdfInvalidError->Error
+    } else if density == 0.0 {
+      infinity->Ok
+    } else {
+      density->logFn->(x => -.x)->Ok
+    }
+  }
+  let scoreWithPrior' = (
+    ~estimatePdf: float => float,
+    ~answer: scalar,
+    ~priorPdf: float => float,
   ): result<float, Operation.Error.t> => {
-    let numerator = answer->predictionPdf
-    if numerator < 0.0 {
+    let numerator = answer->estimatePdf
+    let priorDensityOfAnswer = answer->priorPdf
+    if numerator < 0.0 || priorDensityOfAnswer < 0.0 {
       Operation.PdfInvalidError->Error
-    } else if numerator == 0.0 {
+    } else if numerator == 0.0 || priorDensityOfAnswer == 0.0 {
       infinity->Ok
     } else {
-      -.(
-        switch priorPdf {
-        | None => numerator->logFn
-        | Some(f) => {
-            let priorDensityOfAnswer = f(answer)
-            if priorDensityOfAnswer == 0.0 {
-              neg_infinity
-            } else {
-              (numerator /. priorDensityOfAnswer)->logFn
-            }
-          }
-        }
-      )->Ok
+      minusScaledLogOfQuot(~esti=numerator, ~answ=priorDensityOfAnswer)
     }
   }
+
+  let sum = (mp: PointSetTypes.MixedPoint.t): float => mp.continuous +. mp.discrete
+  let score = (~estimate: t, ~answer: scalar): result<float, Operation.Error.t> => {
+    let estimatePdf = x =>
+      switch estimate {
+      | Continuous(esti) => Continuous.T.xToY(x, esti)->sum
+      | Discrete(esti) => Discrete.T.xToY(x, esti)->sum
+      | Mixed(esti) => Mixed.T.xToY(x, esti)->sum
+      }
+
+    score'(~estimatePdf, ~answer)
+  }
+  let scoreWithPrior = (~estimate: t, ~answer: scalar, ~prior: t): result<
+    float,
+    Operation.Error.t,
+  > => {
+    let estimatePdf = x =>
+      switch estimate {
+      | Continuous(esti) => Continuous.T.xToY(x, esti)->sum
+      | Discrete(esti) => Discrete.T.xToY(x, esti)->sum
+      | Mixed(esti) => Mixed.T.xToY(x, esti)->sum
+      }
+    let priorPdf = x =>
+      switch prior {
+      | Continuous(prio) => Continuous.T.xToY(x, prio)->sum
+      | Discrete(prio) => Discrete.T.xToY(x, prio)->sum
+      | Mixed(prio) => Mixed.T.xToY(x, prio)->sum
+      }
+    scoreWithPrior'(~estimatePdf, ~answer, ~priorPdf)
+  }
 }
+
+module TwoScalars = {
+  let score = (~estimate: scalar, ~answer: scalar) =>
+    if answer == 0.0 {
+      0.0->Ok
+    } else if estimate == 0.0 {
+      infinity->Ok
+    } else {
+      minusScaledLogOfQuot(~esti=estimate, ~answ=answer)
+    }
+
+  let scoreWithPrior = (~estimate: float, ~answer: float, ~prior: float) =>
+    if answer == 0.0 {
+      0.0->Ok
+    } else if estimate == 0.0 || prior == 0.0 {
+      infinity->Ok
+    } else {
+      minusScaledLogOfQuot(~esti=estimate /. prior, ~answ=answer)
```

Comment:
I'm somewhat sure this TwoScalars code is pretty wrong. It seems to me like it's pretending the scoring is done similar to continuous distributions, but at the point in question. But scalar scoring is totally different; the values are on the x-axis, not the y-axis.

Reply:
If the answer is 0.0 and I guess 1.0, I'm very wrong; the score isn't 0.

Reply:
@NunoSempere @OAGr I'd like to split out `ScalarEstimateScalarAnswer` out of the type concerning `logScore` input and into it's own type, where I will implement 1D euclidean distance. What do we think?

Reply:
Maybe have it somewhere else, but still call it from 'log type'? It still would be nice to have a simple interface to everything including float to float; my impression is that 'log type' was supposed to be this interface.

Reply:
No I want 
```rescript
type logScoreArgs =
  | DistEstimateDistAnswer(abstractScoreArgs<pointSetDist, pointSetDist>)
  | DistEstimateScalarAnswer(abstractScoreArgs<pointSetDist, scalar>)
type distanceScoreArgs = 
  | ScalarEstimateScalarAnswer(abstractScoreArgs<scalar, scalar>)
```
and then we can have 
- `logScore : record => float`
- `distanceScore: record => float`
- `score : record => float` which has the ability to dispatch to the previous two.

Reply:
> where I will implement 1D euclidean distance

As opposed to the log distance? I'm pretty sure this wouldn't be a proper scoring rule. Not sure I understand.